app/components/portfolio/SearchResults.tsx

```diff
@@ -0,0 +1,82 @@
+import React, { useEffect, useRef } from "react";
+import Image from "next/image";
+import { useSelector } from "react-redux";
+import { useGetSearchCoinsDataQuery, useGetTenCoinsPricesQuery } from "@/app/lib/marketSlice";
+import { selectDarkmode, selectCurrency } from "@/app/lib/dynamicValuesSlice";
+import clsx from "clsx";
+type Coin = {
+  name: string;
+  id: string;
+  symbol: string;
+  thumb: string;
+};
+const DropdownSearch = ({query, toggleSearch, clearSearch, handleCoin} : {
+    query: string, 
+    toggleSearch: ()=>void, 
+    clearSearch: ()=>void,
+    handleCoin: any
+  }
+) => {
+  const currency = useSelector(selectCurrency);
+  const { data } = useGetSearchCoinsDataQuery(query);
+  const darkmode = useSelector(selectDarkmode);
+  const coinsForRender = data?.coins.slice(0,10);
+  const queryTenCoins = coinsForRender?.map((coin: Coin) => coin.id).join("%2C");
+  const { data: prices } = useGetTenCoinsPricesQuery(queryTenCoins, currency);
+  const pricesStore = {} as {[key: string]: [number, boolean, string]};
+  prices && Object.entries(prices).map((price) => {
+    const coinSymbol = data?.coins.filter((coin: Coin) => coin.id === price[0])[0]?.symbol;
+    const goingUp = prices[price[0]][`${currency.label.toLowerCase()}_24h_change`] > 0;
+    pricesStore[price[0]] = [prices[price[0]][currency.label.toLowerCase()], goingUp, coinSymbol];
+  });
+  const ref = useRef<HTMLDivElement>(null);
+  const handleClickCoin = (e: React.MouseEvent<HTMLDivElement>, coin: Coin) => {
+    e.preventDefault();
+    handleCoin(coin);
+    toggleSearch();
+    clearSearch();
+  };
+  useEffect(() => {
+    const handleClickOutside = (event: MouseEvent) => {
+      if (ref.current && !ref.current.contains(event.target as Node)) {
+        toggleSearch();
+        clearSearch();
+      }
+    };
+    document.addEventListener("mousedown", handleClickOutside);
+    return () => {
+      document.removeEventListener("mousedown", handleClickOutside);
+    };
+  },[clearSearch, toggleSearch]);
```

Comment:
I would extract this clickoutside logic and put it into a custom hook in the next PR

Reply:
I agree with suggestion